mlrun/api/schemas/model_endpoints.py

```diff
@@ -1,106 +1,232 @@
-                "and ModelEndpoint.metadata.tag must be initalized"
+                "Both function_uri and versioned_model have to be initialized"
             )
 
-        endpoint_unique_string = (
-            f"{self.spec.function}_{self.spec.model}_{self.metadata.tag}"
+        uid = ModelEndpoint.EndpointUID(
+            function_uri.project,
+            function_uri.function,
+            function_uri.tag,
+            versioned_model.model,
+            versioned_model.version,
         )
-        md5_str = md5(endpoint_unique_string.encode("utf-8")).hexdigest()
-        return f"{self.metadata.project}.{md5_str}"
 
+        return uid
 
-class Histogram(BaseModel):
-    buckets: List[Tuple[float, float]]
-    count: List[int]
+    @dataclass
+    class FunctionURI:
+        project: str
+        function: str
+        tag: Optional[str] = None
 
+        @classmethod
+        def from_string(cls, function_uri):
+            project, function_with_tag = function_uri.split("/")
 
-class Metric(BaseModel):
-    name: str
-    start_timestamp: str
-    end_timestamp: str
-    headers: List[str]
-    values: List[Tuple[str, float]]
-    min: float
-    avg: float
-    max: float
+            try:
+                function, tag = function_with_tag.split(":")
+            except ValueError:
+                function, tag = function_with_tag, None
 
+            return cls(project, function, tag)
 
-class FeatureValues(BaseModel):
-    min: float
-    avg: float
-    max: float
-    histogram: Histogram
+    @dataclass
+    class VersionedModel:
+        model: str
+        version: Optional[str]
 
+        @classmethod
+        def from_string(cls, model):
+            try:
+                model, version = model.split(":")
+            except ValueError:
+                model, version = model, None
 
-class Features(BaseModel):
-    name: str
-    weight: float
-    expected: FeatureValues
-    actual: Optional[FeatureValues]
+            return cls(model, version)
 
+    @dataclass
+    class EndpointUID:
+        project: str
+        function: str
+        function_tag: str
+        model: str
+        model_version: str
+        uid: Optional[str] = None
 
-class ModelEndpointState(BaseModel):
-    endpoint: ModelEndpoint
-    first_request: Optional[str] = None
-    last_request: Optional[str] = None
-    accuracy: Optional[float] = None
-    error_count: Optional[int] = None
-    drift_status: Optional[str] = None
-    metrics: Dict[str, Metric] = {}
-    features: List[Features] = []
+        def __post_init__(self):
+            versioned_function = f"{self.function}_{self.function_tag or 'N/A'}"
+            versioned_model = f"{self.model}_{self.model_version or 'N/A'}"
+            unique_string = f"{versioned_function}_{versioned_model}"
+            md5_str = md5(unique_string.encode("utf-8")).hexdigest()
+            self.uid = f"{self.project}.{md5_str}"
```

Comment:
Why the project isn't in the hex calculation as well ? and then will just have a uid and not some half human readable string half uid

Reply:
As per @yaronha's request